src/models/game_session.ts

```diff
@@ -151,7 +151,15 @@ export default class GameSession {
             await db.kmq("guild_preferences")
                 .where("guild_id", message.guildID)
                 .increment("songs_guessed", 1);
-            this.startRound(guildPreference, message);
+            if (guildPreference.getGoal() == 0 || this.scoreboard.getWinners()[0].getScore() < guildPreference.getGoal()) {
```

Comment:
From previous comment, use the guild preference method to check if a goal is set